### PR TITLE
opt: support new features in SQLite

### PR DESCRIPTION
**Pull request by bajinsheng**
Some new features are required for QPG in SQLite.

* a maximum number of tables/indexes: QPG gradually mutates the database state, so the database state may be extremely sophisticated and should be controlled.
* explain interface: adding an interface to get query plan by a query in string format.
* some actions are incorporated.

### CODE REVIEW

File: src/sqlancer/sqlite3/SQLite3Options.java [hunk 1]
```diff
@@ -90,6 +90,13 @@ public class SQLite3Options implements DBMSSpecificOptions<SQLite3OracleFactory>
             "--execute-queries" }, description = "Specifies whether the query in the fuzzer should be executed", arity = 1)
     public boolean executeQuery = true;
 
+    @Parameter(names = {
+            "--max-tables" }, description = "The maximum number of tables/virtual tables/ rtree tables/ views that can be created")
```
Comment (mrigger) [inline src/sqlancer/sqlite3/SQLite3Options.java:94]: Could we use `--max-num-tables` here and `--max-num-indexes` below, to match the naming convention of other option classes (see below)?

https://github.com/sqlancer/sqlancer/blob/a535520766808fd5bb71db41a0351e9d1f0d5331/src/sqlancer/duckdb/DuckDBOptions.java#L84

File: src/sqlancer/sqlite3/SQLite3Options.java [hunk 2]
```diff
@@ -90,6 +90,13 @@ public class SQLite3Options implements DBMSSpecificOptions<SQLite3OracleFactory>
...
+    public int maxTable = 10;
```
Comment (mrigger) [inline src/sqlancer/sqlite3/SQLite3Options.java:95]: perhaps `maxNumTables`?

File: src/sqlancer/sqlite3/SQLite3Options.java [hunk 3]
```diff
@@ -90,6 +90,13 @@ public class SQLite3Options implements DBMSSpecificOptions<SQLite3OracleFactory>
...
+
+    @Parameter(names = { "--max-index" }, description = "The maximum number of indexes that can be created")
+    public int maxIndex = 20;
```
Comment (mrigger) [inline src/sqlancer/sqlite3/SQLite3Options.java:98]: Perhaps `maxNumIndexes`?

File: src/sqlancer/sqlite3/SQLite3Options.java (hunk 1 above)
Comment (bajinsheng) [reply src/sqlancer/sqlite3/SQLite3Options.java]: Fixed

File: src/sqlancer/sqlite3/SQLite3Provider.java [hunk 4]
```diff
@@ -58,47 +58,50 @@ public SQLite3Provider() {
...
 
     public enum Action implements AbstractAction<SQLite3GlobalState> {
-        PRAGMA(SQLite3PragmaGenerator::insertPragma), //
-        INDEX(SQLite3IndexGenerator::insertIndex), //
-        INSERT(SQLite3InsertGenerator::insertRow), //
-        VACUUM(SQLite3VacuumGenerator::executeVacuum), //
-        REINDEX(SQLite3ReindexGenerator::executeReindex), //
-        ANALYZE(SQLite3AnalyzeGenerator::generateAnalyze), //
-        DELETE(SQLite3DeleteGenerator::deleteContent), //
+        PRAGMA(SQLite3PragmaGenerator::insertPragma), // 0
```
Comment (mrigger) [inline src/sqlancer/sqlite3/SQLite3Provider.java:61]: Not sure why we are numbering the actions here?

**Review by mrigger — APPROVED** [review]
LGTM

File: src/sqlancer/sqlite3/SQLite3Provider.java (hunk 4 above)
Comment (bajinsheng) [reply src/sqlancer/sqlite3/SQLite3Provider.java:61]: I number these actions for debugging. The enumeration variable only includes index information in execution, and these numbers help me to quickly locate which action is selected and executed.

**Pull request merged**